samples/matrix_multiply/matrix_multiply.cpp

```diff
@@ -271,65 +298,90 @@ int main(int argc, char *argv[]) {
-       * some error occurs while the queue is executing kernels) and one of
-       * cl::sycl::queue::throw() or cl::sycl::queue::wait_and_throw() is
-       * called. */
-      queue q([&](exception_list eL) {
-        try {
-          for (auto &e : eL) {
+      {
+        /* Create the SYCL queue - note that we add an async handler function
+         * to capture potential asynchronous errors. This function will be
+         * called every time there is an asynchronous error on the queue (i.e.
+         * some error occurs while the queue is executing kernels) and one of
+         * cl::sycl::queue::throw() or cl::sycl::queue::wait_and_throw() is
+         * called. */
+        queue q([&](exception_list eL) {
+            try {
+            for (auto &e : eL) {
             std::rethrow_exception(e);
-          }
-        } catch (cl::sycl::exception e) {
-          std::cout << " An exception has been thrown: " << e.what()
-                    << std::endl;
-        }
-      });
-
-      auto start = std::chrono::steady_clock::now();
-      local_mxm(q, MA, MB, MC, matSize);
-      auto end = std::chrono::steady_clock::now();
-      std::cout << "SYCL: ";
-      auto time = std::chrono::duration_cast<std::chrono::milliseconds>(
-                      end - start).count();
-      std::cout << "Time: " << time << std::endl;
-      float flops =
+            }
+            } catch (cl::sycl::exception e) {
+            std::cout << " An exception has been thrown: " << e.what()
+            << std::endl;
+            }
+            });
+
+        auto start = std::chrono::steady_clock::now();
+        error = local_mxm(q, MA, MB, MC, matSize);
+        q.wait_and_throw();
+        auto end = std::chrono::steady_clock::now();
+        auto time = std::chrono::duration_cast<std::chrono::milliseconds>(
+            end - start).count();
+        std::cout << "SYCL: ";
+        std::cout << "Time: " << time << std::endl;
+        float flops =
           (2.0f * matSize * matSize * matSize / (time / 1000.0f)) * 1.0e-9f;
-      std::cout << "GFLOPs: " << flops << std::endl;
-    }
-  }
+        std::cout << "GFLOPs: " << flops << std::endl;
+        std::cout << " Output " << std::endl;
+      }
 
-  std::cout << " Output " << std::endl;
-  display_matrix(MC, matSize);
-  bool error = false;
-  // Testing
-  for (int i = 0; i < matSize; i++)
-    for (int j = 0; j < matSize; j++) {
-      if (std::fabs(MC[i * matSize + j] - MB[i * matSize + j]) > 1e-8) {
-        std::cout << " Position " << i << ", " << j
-                  << " differs: " << MC[i * matSize + j]
-                  << " != " << MB[i * matSize + j] << std::endl;
-        error = true;
+      if (!error) {
+        display_matrix(MC, matSize);
+        error = false;
+        // Testing
+        for (int i = 0; i < matSize; i++)
+          for (int j = 0; j < matSize; j++) {
+            if (std::fabs(MC[i * matSize + j] - MB[i * matSize + j]) > 1e-8) {
```

Comment:
Can we replace this with something like the sample from here? Sounds pedantic, but I'm worried that for larger numbers (say ~100) that will be spuriously false, or that for small numbers it will be incorrectly true.